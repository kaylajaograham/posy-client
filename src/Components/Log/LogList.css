
.LogList {
    background-color: #efefef;
    height: 100%;
    padding: 10px;
    width: 100%;
  }
  
  .LogList__content {
    color: goldenrod;
    max-width: 380px;
  }
  