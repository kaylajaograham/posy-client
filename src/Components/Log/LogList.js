import React from 'react'

import './LogList.css';

import PosyContext from '../PosyContext'
import Log from '.Log';
import { findLog } from '../../log-helpers';

export default class LogList extends React.Component {
  state = {
    forErrors: this.props.match,
    toggle: true
  }

  static defaultProps = {
    match: {
      params: {}
    }
  }

  static contextType = PosyContext;

  // sends user back to homepage after delete
  handleDeleteLog = logId => {
    this.props.history.push('/')
  }
    

  render () {
    const { logs=[] } = this.context
    const { logId } = this.state.forErrors.params
    const log = findLog(logs, logId) || { content: ''}
      if(this.state.toggle === false) {
        this.setState({
          forErrors: 'err'
        })
        this.setState({
          forErrors: this.props.match
        })
      }

    return (
      <section className='LogList'>
        <Log
          id={log.id}
          log_name={log.log_name}
          log_tag={log.log_tag}
          log_entry={log.log_entry}
          onDeleteLog={this.handleDeleteLog}
        />
        <div className='LogList__content'>
          {log.content.split(/\n \r|\n/).map((para, i) =>
            <p key={i}>{para}</p>
          )}
        </div>
      </section>
    )    
  }

}
