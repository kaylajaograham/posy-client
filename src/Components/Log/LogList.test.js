import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LogList from './LogList';

describe(`ItineraryList component`, () => {
  const props = {
    itinerary: {
      "id": 1,
      "title": "test"
    }
  };

  it('renders the ItineraryList given props', () => {
    const wrapper = shallow(<LogList {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})