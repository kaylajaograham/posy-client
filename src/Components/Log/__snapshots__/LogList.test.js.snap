// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItineraryList component renders the ItineraryList given props 1`] = `
<section
  className="LogList"
>
  <div
    className="LogList__content"
  />
</section>
`;
